forms/Form.php

```diff
@@ -552,13 +578,13 @@ public function getRedirectToFormOnValidationError() {
 	/**
 	 * Add a plain text error message to a field on this form.  It will be saved into the session
 	 * and used the next time this form is displayed.
+	 *
+	 * @deprecated 3.2
 	 */
-	public function addErrorMessage($fieldName, $message, $messageType, $escapeHtml = true) {
-		Session::add_to_array("FormInfo.{$this->FormName()}.errors",  array(
-			'fieldName' => $fieldName,
-			'message' => $escapeHtml ? Convert::raw2xml($message) : $message,
-			'messageType' => $messageType,
-		));
+	public function addErrorMessage($fieldName, $message, $messageType) {
+		Deprecation::notice('3.2', 'Throw a ValidationException instead.');
```

Comment:
Is this really the only possible option? What if you want to add several errors and then do some custom routing with the resulting form? Couldn't the deprecation notice ask the user to use getSessionValidationResult and add the error there instead?